examples/mediadevices/src/helpers.js

```diff
@@ -51,19 +51,12 @@ function applyAudioOutputDeviceSelection(deviceId, audio) {
  * @param {Room} [room] - The Room, if you have already joined one
  * @returns {Promise<void>}
  */
-function applyAudioInputDeviceSelection(deviceId, audio, room) {
-  return Video.createLocalAudioTrack({
-    deviceId: {
-      exact: deviceId // NOTE: on ios safari - it respects the deviceId only if its exact.
-    }
-  }).then(function(localTrack) {
-    localTrack.attach(audio);
-    if (room) {
-      switchLocalTracks(room, localTrack);
-    }
-  }).catch(function(error) {
-    console.log('applyAudioInputDeviceSelection failed:', error);
-  });
+function applyAudioInputDeviceSelection(localAudioTrack, audio, room) {
```

Comment:
@PikaJoyce I would prefer if we kept the track creation within the helper function:
```js
function applyAudioInputDeviceSelection(deviceId, audio, room) {
  // Stop and unpublish the existing LocalAudioTrack.
  if (room) {
    room.localParticipant.audioTracks.forEach(function(publication) {
      publication.track.stop();
      publication.unpublish();
    });
  }
  // Create the new LocalAudioTrack and publish it to the Room.
  return Video.createLocalAudioTrack({
    deviceId: {
      exact: deviceId // NOTE: on ios safari - it respects the deviceId only if its exact.
    }
  }).then(function(localTrack) {
    localTrack.attach(audio);
    if (room) {
      return room.localParticipant.publishTrack(localTrack);
    }
  }).catch(function(error) {
    console.log('applyAudioInputDeviceSelection failed:', error);
  });
}
```
Also, I don't think we need the `switchLocalTracks` function anymore.

Reply:
Sure. I can make those changes. But I don't have a webcam to test unfortunately.

But I'll work on it!